File: app/domain/models/tool_call.py
"""ToolCall 审计领域模型（Phase 1）。"""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any


@dataclass
class ToolCall:
    """工具调用审计记录（只追加写，不可变）。"""
    id: str
    session_id: str
    task_id: str
    agent_id: str
    tool_name: str
    status: str                        # RUNNING | SUCCEEDED | FAILED
    arguments: dict[str, Any] = field(default_factory=dict)
    result: str | None = None
    error: str | None = None
    started_at: str = ""
    finished_at: str = ""

    def to_dict(self) -> dict[str, Any]:
        return {
            "id": self.id,
            "session_id": self.session_id,
            "task_id": self.task_id,
            "agent_id": self.agent_id,
            "tool_name": self.tool_name,
            "status": self.status,
            "arguments": self.arguments,
            "result": self.result,
            "error": self.error,
            "started_at": self.started_at,
            "finished_at": self.finished_at,
        }

    @classmethod
    def from_dict(cls, d: dict[str, Any]) -> "ToolCall":
        return cls(
            id=d["id"],
            session_id=d["session_id"],
            task_id=d["task_id"],
            agent_id=d["agent_id"],
            tool_name=d["tool_name"],
            status=d["status"],
            arguments=d.get("arguments", {}),
            result=d.get("result"),
            error=d.get("error"),
            started_at=d.get("started_at", ""),
            finished_at=d.get("finished_at", ""),
        )

File: miniagents-assembled/src/main/java/com/codex/miniagents/domain/model/tool/ToolCall.java
package com.codex.miniagents.domain.model.tool;

import lombok.Getter;

import java.time.Instant;
import java.util.HashMap;
import java.util.Map;

@Getter
public class ToolCall {
    private String id;

    private String sessionId;

    private String taskId;

    private String agentId;

    private String toolName;

    private String status;

    private Map<String, Object> arguments = new HashMap<>();

    private String result;

    private String error;

    private Instant startedAt;

    private Instant finishedAt;

    public ToolCall() {
    }

    public ToolCall(String id, String sessionId, String taskId, String agentId, String toolName, String status,
        Map<String, Object> arguments, String result, String error, Instant startedAt, Instant finishedAt) {
        this.id = id;
        this.sessionId = sessionId;
        this.taskId = taskId;
        this.agentId = agentId;
        this.toolName = toolName;
        this.status = status;
        this.arguments = arguments == null ? new HashMap<>() : new HashMap<>(arguments);
        this.result = result;
        this.error = error;
        this.startedAt = startedAt;
        this.finishedAt = finishedAt;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setSessionId(String sessionId) {
        this.sessionId = sessionId;
    }

    public void setTaskId(String taskId) {
        this.taskId = taskId;
    }

    public void setAgentId(String agentId) {
        this.agentId = agentId;
    }

    public void setToolName(String toolName) {
        this.toolName = toolName;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public void setArguments(Map<String, Object> arguments) {
        this.arguments = arguments == null ? new HashMap<>() : new HashMap<>(arguments);
    }

    public void setResult(String result) {
        this.result = result;
    }

    public void setError(String error) {
        this.error = error;
    }

    public void setStartedAt(Instant startedAt) {
        this.startedAt = startedAt;
    }

    public void setFinishedAt(Instant finishedAt) {
        this.finishedAt = finishedAt;
    }
}
